test/domains/userAPI/test2DdomainOrderToIndex.chpl

```diff
@@ -0,0 +1,29 @@
+proc testDom(dom: domain) {
+  for (idx, i) in zip(dom, 0..<dom.size) {
+    if idx != dom.orderToIndex(i) {
+      writeln('wrong index for ', dom, ':', idx, ' != dom.orderToIndex(', i,')');
+      writeln('  got ', dom.orderToIndex(i), ' instead');
+    }
+  }
+}
+
+{
+    var dom = {1..3,1..3,1..5};
+    testDom(dom);
+}
+{
+    var dom = {1..3,1..3,1..5 by -2};
+    testDom(dom);
+}
+{
+    var dom = {1..4 by 2,1..4 by 2,1..4 by -2};
+    testDom(dom);
+}
+{
+    var dom = {1..4 by 2,1..4 by 2 align 2,1..4 by -2};
+    testDom(dom);
+}
+{
+    var dom = {1..3,1..3};
+    testDom(dom);
+}
```

Comment:
It would be a good idea to add the specific examples from the documentation into this test file to ensure future changes don't break the documentation examples. We can annotate these test cases with a reminder of why they are here:

```chpl
{
  // Example mentioned in the documentation
  var dom = {2..10 by 2};
  const idx = 4;
  const i = 2;
  if idx != dom.orderToIndex(i) {
    writeln('wrong index for ', dom, ':', idx, ' != dom.orderToIndex(', i,')');
    writeln('  got ', dom.orderToIndex(i), ' instead');
  }
}

{
  // <other example here>
}
```